src/utils/media.ts

```diff
@@ -7,31 +8,31 @@ type DeprecatedPlaylistItem = {
   seriesPlaylistId?: string;
 };
 
-export const getSeriesId = (item: PlaylistItem & DeprecatedPlaylistItem) => {
-  if (!item) {
-    return undefined;
-  }
-
-  return item['seriesPlayListId'] || item.seriesPlaylistId || item.seriesId;
-};
-
 export const isPlaylist = (item: unknown): item is Playlist => !!item && typeof item === 'object' && 'feedid' in item;
 export const isPlaylistItem = (item: unknown): item is PlaylistItem => !!item && typeof item === 'object' && 'mediaid' in item;
 
-export const isSeriesPlaceholder = (item: PlaylistItem) => {
-  return typeof getSeriesId(item) !== 'undefined';
+// For the deprecated series flow we store seriesId in custom params
+export const getSeriesPlaylistIdFromCustomParams = (item: PlaylistItem & DeprecatedPlaylistItem) =>
+  item.seriesPlayListId || item.seriesPlaylistId || item.seriesId;
+
+// For the deprecated flow we store seriesId in the media custom params
+export const isDeprecatedSeriesFlow = (item: PlaylistItem) => {
+  return typeof getSeriesPlaylistIdFromCustomParams(item) !== 'undefined';
 };
 
+// For the new series flow we use contentType custom param to define media item to be series
+// In this case media item and series item have the same id
+export const isNewSeriesFlow = (item: PlaylistItem) => item.contentType === CONTENT_TYPE.series;
+
+export const isSeries = (item: PlaylistItem) => isDeprecatedSeriesFlow(item) || isNewSeriesFlow(item);
+
 export const isEpisode = (item: PlaylistItem) => {
-  return item && typeof item.episodeNumber !== 'undefined';
+  return typeof item?.episodeNumber !== 'undefined' || item?.contentType === CONTENT_TYPE.episode;
```

Comment:
Seeing this made me think of a question - does this mean that all of the series functionality relies on every item inside a series being assigned content_type = episode?

That doesn't happen automatically does it? That seems very limiting... is there a way we can specify URLs so that as long as an episode is accessed through a series it will show up as an episode regardless of content type?  Should we always check if a media item belongs to a series before routing it by content type?

Reply:
We also rely on the content type (or custom params) when working with series and live channels. That is how screen router works now. You have to manually set content type using dashboard before.

We could check it in `MediaScreenRouter` but this logic looks a bit out of the context. Maybe we could also add content type `episode` in case we are sure it has a parent series object. 

```
const MediaScreenRouter = () => {
  const params = useParams();
  const id = params.id || '';
  const { isLoading, isFetching, error, data } = useMedia(id);
  const { data: episode } = useEpisode(id);

  const { t } = useTranslation('error');

  if (isLoading) {
    return <Loading />;
  }

  if (error || !data) {
    return <ErrorPage title={t('video_not_found')} />;
  }

  if (episode) {
    data.contentType = 'episode'
  }

  const MediaScreen = mediaScreenMap.getScreen(data);

  return <MediaScreen data={data} isLoading={isFetching} />;
};
```

Reply:
Assigning the series content type happens 'automatically' now by creating a series in the media library, but assigned 'episode' types does not, so I don't think we can rely on it in the same way. Personally, I would suggest we achieve this by routing, either use the `/s/` routes, add an `/e/` route, or just use custom params. This way we can make sure that media items belonging to a series which are accessed through a series display properly, but it does mean they will be treated as normal media items if they are accessed directly through a playlist or something unless they are marked with content type.